components-api/src/main/scala/pl/touk/nussknacker/engine/api/json/encoders/ToJsonEncoder.scala

```diff
@@ -0,0 +1,72 @@
+package pl.touk.nussknacker.engine.api.json.encoders
+
+import cats.data.Validated.Valid
+import cats.data.ValidatedNel
+import io.circe.Json
+import io.circe.Json._
+import pl.touk.nussknacker.engine.util.json.ToJsonEncoderCustomisation
+
+import java.util.ServiceLoader
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+object ToJsonEncoder {
+
+  // The default behaviour is to fail-fast. This approach helps in an investigation of a potential wrong type usage in the json context.
+  val default: StrictToJsonEncoder.type = StrictToJsonEncoder
+
+  // For this encoder, encodeUnsafe() doesn't throw exceptions for unknown values - as a fallback .toString method is used
+  // encode() method is hidden because it always returns Valid
+  val looseEncoder = new ToJsonEncoder(failOnUnknown = false, classOf[ToJsonEncoder].getClassLoader)
+
+}
+
+object StrictToJsonEncoder extends ToJsonEncoder(failOnUnknown = true, classOf[ToJsonEncoder].getClassLoader) {
+  // Unhides encode method, which is hidden for loose encoder
```

Comment:
It is not really important nitpick, but maybe for better readability underlying method could be changed to `protected def doEncode`, and this method would just invoke it, without unhiding/using `super.` etc..